### PR TITLE
added custom metrics for read scenario

### PR DESCRIPTION
**Pull request by vlad-diachenko**
implemented logic to report custom k6 metrics based on response stats for instant and range queries

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -25,6 +28,11 @@ const (
 	TenantPrefix = "xk6-tenant"
 )
 
+var (
+	BytesProcessedPerSecondsSummary = k6_stats.New("loki_bytes_precessed_per_second_summary", k6_stats.Trend, k6_stats.Data)
+	LinesProcessedPerSecondsSummary = k6_stats.New("loki_lines_precessed_per_second_summary", k6_stats.Trend, k6_stats.Default)
```
Comment (chaudum) [inline client.go:33]: Why call it "summary"?

Comment (chaudum) [reply client.go:33]:
```suggestion
	BytesProcessedPerSecond = k6_stats.New("loki_bytes_processed_per_second", k6_stats.Trend, k6_stats.Data)
	LinesProcessedPerSecond = k6_stats.New("loki_lines_processed_per_second", k6_stats.Trend, k6_stats.Default)
```

File: client.go [hunk 2]
```diff
@@ -250,7 +266,43 @@ func (c *Client) send(ctx context.Context, state *lib.State, buf []byte, useProt
...
+		Samples: []k6_stats.Sample{
+			{
+				Metric: BytesProcessedPerSecondsSummary,
+				Tags:   k6_stats.NewSampleTags(map[string]string{"endpoint": queryType.Endpoint()}),
+				Value:  float64(responseWithStats.Data.Stats.Summary.BytesProcessedPerSecond),
+				Time:   now,
+			},
+			{
+				Metric: LinesProcessedPerSecondsSummary,
+				Tags:   k6_stats.NewSampleTags(map[string]string{"method": queryType.Endpoint()}),
```
Comment (chaudum) [inline client.go:301]: We should use the same tags in both metrics.

```suggestion
				Tags:   k6_stats.NewSampleTags(map[string]string{"endpoint": queryType.Endpoint()}),
```

**Comment by chaudum** [comment]
Can you please also add the metrics (+ the push metrics) to the README?

File: client.go (hunk 1 above)
Comment (vlad-diachenko) [reply client.go]: done

File: client.go (hunk 2 above)
Comment (vlad-diachenko) [reply client.go]: done

**Comment by vlad-diachenko** [comment]
> Can you please also add the metrics (+ the push metrics) to the README?

@chaudum  done

**Review by chaudum — COMMENTED** [review]
Have you also thought about adding `TotalBytesProcessed` and `TotalLinesProcessed` from the response summary?

**Comment by vlad-diachenko** [comment]
> `TotalLinesProcessed` fro

It depends on customer needs... `bytesProcessedPerSecond` and `linesProcessedPerSecond` show us the speed of the Loki. `TotalBytesProcessed` and `TotalLinesProcessed` show the only amount of data that is processed during the query. 
If you think that these two fields are useful, we can add them. 
What do you think?

File: README.md [hunk 3]
```diff
@@ -165,6 +165,26 @@ Execute a series query ([GET /loki/api/v1/series](https://grafana.com/docs/loki/
...
+The extension collects metrics that are printed in the
+[end-of-test summary](https://k6.io/docs/results-visualization/end-of-test-summary/) in addition to the built-in metrics.
+
+### Query metrics
+These metrics are collected only for instant and range queries.
+
+| name | description                                 |
+| ---- |---------------------------------------------|
+| `loki_bytes_precessed_per_second_summary` | Total of bytes processed by Loki per second |
+| `loki_lines_precessed_per_second_summary` | Total lines processed by Loki per second    |
```
[the remark this answers is not in the document]
Comment (vlad-diachenko) [reply README.md:179]: done

File: client.go [hunk 4]
```diff
@@ -25,6 +28,11 @@ const (
 	TenantPrefix = "xk6-tenant"
 )
 
+var (
+	BytesProcessedPerSeconds = k6_stats.New("loki_bytes_precessed_per_second", k6_stats.Trend, k6_stats.Data)
+	LinesProcessedPerSeconds = k6_stats.New("loki_lines_precessed_per_second", k6_stats.Trend, k6_stats.Default)
```
[the remark this answers is not in the document]
Comment (vlad-diachenko) [reply client.go:33]: done

**Comment by chaudum** [comment]
> If you think that these two fields are useful, we can add them.

Yes, I think both metrics are useful.

**Review by chaudum — APPROVED** [review]

**Comment by vlad-diachenko** [comment]
thanks @chaudum

**Pull request merged**